### ISSUE TITLE
Improve exception handling with InvocationTargetException

### ISSUE THREAD

**Issue by No3x** [seaside1/jrule#15]
Currently some exceptions are catched if the invoked method fails but I see only
`2022-01-05 23:36:04.013 [ERROR] [on.jrule.internal.engine.JRuleEngine] - [UpdateWeightRule] Error` in the openhab log. Which is not helpful.
https://github.com/seaside1/jrule/blob/39b467cbdac0b754061d11f09b532a984c96557c/src/main/java/org/openhab/automation/jrule/internal/engine/JRuleEngine.java#L467-L471
I guess there is a [InvocationTargetException](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/reflect/InvocationTargetException.html) caused in some internal classes that are called by a rule of mine. 
In case of an `InvocationTargetException` the actual cause of the exception has to be unwrapped via `getCause()`. Please see linked javadoc as reference.
Can you please add something like:
```java
catch (InvocationTargetException ite) {
    JRuleLog.error(logger, context.getRuleName(), "Error", ite.getCause()); 
}
```
or maybe `"Error {}"` also helps.

**Comment by No3x** [comment]
I'm going to test this and contribute.
